### ISSUE TITLE
SteamOS in-home streaming stuttering 

### ISSUE THREAD

**Issue by Mohandevir** [alkazar/steamos-compositor#6]
Thanks for your SteamOS-Compositor support.

I have a question: 
Is there anything that can be done to prevent steam in-home streaming from becoming a stuttering fest when streaming from a SteamOS host?

When I use the desktop client with Ubuntu, in-home streaming performances are awesome, but, if possible, I would like to build a "SteamOS server" for my Nvidia Shield.

Thanks again.

**Comment by alkazar** [comment]
I am not sure what the cause could be. It is doubtful the issue is with the compositor though.

You might try running the SteamOS compositor on Ubuntu to make sure. You could use https://github.com/ShadowApex/steamos-ubuntu to do that.

You could also try my SteamOS alternative: https://github.com/alkazar/gamer-os

**Comment by Mohandevir** [comment]
Thanks for the quick answer.

I tried SteamOS with steamos-compositor, Ubuntu with steamos-compositor, Ubuntu vanilla DE, Ubuntu Studio DE, KDE and Mate. The only common point where the stuttering is present is when I use steamos-compositor, no matter the underlying OS. I don't know what may be the cause, but these are my observations.

I even tried modifying my Nvidia drivers parameters (allow flipping, Force Full Composition Pipeline, etc...) and the Steam streaming parameters, but I couldn't find anything that could solve the issue.

Didn't try GamerOS though, since it's Arch based and, up to now, I tried sticking to debian based distro to stay as close as possible to Valve's officially supported OS. This said, it's probably a great OS, I might try it, eventually.

BTW, removing the colored frame when starting a Proton title is a nice touch... It feels less like a beta now. :) 

Thanks again.
Keep doing your great work.

**Comment by alkazar** [comment]
I see, that certainly seems like the compositor is the cause. I have used SteamOS to stream for quite a while without issues although I haven't tried it recently.

The only thing that comes to mind is the built in vsync in the compositor that might be interfering with streaming somehow.

Sounds like gamerOS wouldn't help you.

I'll try to take a look, but doubt I'll be able to figure anything out. You should create a ticket on the SteamOS github page if you havent already.

**Comment by Mohandevir** [comment]
Ttmo directed me to your page saying that brewmaster's steamos-compositor
is low on their priorities.

This said you make me think that I did have a good streaming experience
with older steamos-compositor versions. It might have been induced with
version 1.35? Maybe I should test with previous builds?

Edit: steamos-compositor-1.25 doesn't want to install because it depends on libudev0...

Thanks anyway for your support.

**Comment by alkazar** [comment]
Interesting, sounds like they might have something new for clockwerk?

**Comment by Mohandevir** [comment]
Lol! I had the same reaction. Crossing my fingers.

Edit: Is there a way to disable the steamos-compositor vsync and let the nvidia driver manage it? I could test it on my Ubuntu build.

**Comment by alkazar** [comment]
I do not know of a way to disable the steamos compositor vsync, sorry.

I did find this though: https://steamcommunity.com/groups/steamuniverse/discussions/1/648814841992270182/

It seems if you add the `-u` option to the compositor, it will do fullscreen unredirection. Might be something to try, but I have no idea if it will help.

**Comment by Mohandevir** [comment]
In the steamos-session script?

**Comment by alkazar** [comment]
Yes

**Comment by Mohandevir** [comment]
Tried it... Streaming stopped working. The host plays the game but the
client shows a black screen.

**Comment by alkazar** [comment]
Closing due to inactivity, lack of reproducability, and unknown nature of the problem.

**Comment by Mohandevir** [comment]
Just an update... I discovered that the Nvidia control panel is accessible on GamerOS. I fired it up and disabled the "Allow Flipping" option. It seems to solve the stuttering while streaming, but it causes tearing in live gaming... That's my conclusion when I tried a couple of games, in both scenarios. I didn't do extensive testing though.

Is there anything that can be done to turn off the "Allow Flipping" when streaming is detected and turn it back on when the streaming session is ended?

Thanks anyway! You're all doing an aswesome work.
